app/src/panels/label/panel-label.vue

```diff
@@ -1,16 +1,99 @@
 <script setup lang="ts">
-withDefaults(
+import { onMounted, onBeforeUnmount, onUpdated, ref } from 'vue';
+import { useAutoFontFit } from '@/composables/use-auto-fit-text';
+
+const props = withDefaults(
 	defineProps<{
 		showHeader?: boolean;
 		text?: string;
+		whiteSpace?: string | undefined;
 		color?: string | undefined;
+		textAlign?: string | undefined;
+		fontSize?: 'sm' | 'md' | 'lg' | 'auto';
+		fontWeight?: number | undefined;
+		fontStyle?: string | undefined;
+		font?: 'sans-serif' | 'serif' | 'monospace';
 	}>(),
 	{
 		showHeader: false,
 		text: '',
 		color: undefined,
+		whiteSpace: 'normal',
+		textAlign: 'center',
+		fontSize: 'auto',
+		fontWeight: 800,
+		fontStyle: undefined,
+		font: 'sans-serif',
 	}
 );
+
+const labelContainer = ref<HTMLDivElement | null>(null);
+const labelText = ref<HTMLParagraphElement | null>(null);
+
+const { adjustFontSize } = useAutoFontFit(labelContainer, labelText);
+
+let resizeObserver: ResizeObserver | null = null;
+
+function adjustPadding() {
+	const container = labelContainer.value;
+	if (!container) return '0px';
+
+	const paddingWidth = container.offsetWidth * 0.05;
+	const paddingHeight = container.offsetHeight * 0.05;
+
+	const padding = Math.round(Math.max(12, Math.min(paddingWidth, paddingHeight)));
+
+	if (props.showHeader == true) {
+		container.style.padding = '0px 12px 12px 12px';
+	} else {
+		container.style.padding = `${padding}px`;
+	}
+
+	return;
+}
+
+function updateFit() {
+	adjustPadding();
+	adjustFontSize();
+
+	if (!resizeObserver) {
+		const container = labelContainer.value;
+		if (!container) return;
+
+		// Create a ResizeObserver to watch for changes in the container's dimensions
+		resizeObserver = new ResizeObserver(() => {
+			updateFit();
+		});
+
+		resizeObserver.observe(container);
+	}
+
+	adjustFontSize();
+}
+
+onMounted(() => {
+	if (props.fontSize == 'auto') {
+		updateFit();
+
+		// Delay the initial font size adjustment to allow the text/font to render fully
+		setTimeout(() => {
+			updateFit();
+		}, 500);
```

Comment:
This feels brittle to me. Any way we can make this event driven?

Reply:
Are we just waiting for Vue's next tick? In that case it might be worth awaiting nextTick instead

Reply:
This was added in because during my testing, I could not get it to consistently do the initial updateFit on load/mount/render of the text. The text needs to be fully rendered, wrapped, and displayed for updateFit to know how much to shrink/grow to make it fit right because it compares the parent's height to the child's height and shrinks it until it fits just right.

Reply:
@br41nslug Could you take a look?

Reply:
@ConnorSimply can you help me reproduce the issue you mentioned running into during testing? In my tests so far with the label panel i cannot see a noticeable difference between with this `setTimeout` and without (commented out) 🤔 Tested in both Firefox and Chrome with and without throtling

Reply:
IMO, this should be enough:
```ts
function updateFit() {
	adjustPadding();
	adjustFontSize();
}

onMounted(() => {
	if (props.fontSize === 'auto') {
		const container = labelContainer.value;
		if (!container) return;

		// Create a ResizeObserver to watch for changes in the container's dimensions
		resizeObserver = new ResizeObserver(() => {
			updateFit();
		});

		resizeObserver.observe(container);
	}
});

onUpdated(() => {
	if (props.fontSize === 'auto') {
		updateFit();
	}
});
```

Note: According to https://stackoverflow.com/a/60026394/20504834 `ResizeObserver` fires initially, which matches my observation in Firefox.

Reply:
@br41nslug Yes, you can reproduce the issue by putting like 1-2 paragraphs in the label for the text and setting the font to serif. When the font is not san-serif the text will resize successfully but it resizes BEFORE the font loads so the size that it resizes to is for sans-serif not serif.

@paescuj Yes, resizeObserver does fire initially. That is not the issue, the issue is that it fires too early before the font finishes loading. My testing locally has the font loading in roughly 200ms after the panel data loads so I figured that 500ms would be a reasonable amount of time to delay a "double check" resize incase fonts are not finished loading.

Reply:
I see! So could we rely on  the https://developer.mozilla.org/en-US/docs/Web/API/FontFaceSet/ready event?

Reply:
That did the trick! :)

Reply:
> Yes, you can reproduce the issue by putting like 1-2 paragraphs in the label for the text and setting the font to serif. When the font is not san-serif the text will resize successfully but it resizes BEFORE the font loads so the size that it resizes to is for sans-serif not serif.

Seems it was fixed with paescuj's changes as i am unable to reproduce the issue on the current state of the branch